test/test_script.py

```diff
@@ -0,0 +1,87 @@
+from collections.abc import Iterator
+from pathlib import Path
+import re
+import subprocess
+
+from dsp_tools import excel2xml
+from lxml import etree
+import pytest
+
+import import_script
+
+
+@pytest.fixture(scope="module")
+def generated_xml_file() -> Iterator[Path]:
+    """Yield the generated XML file as fixture, and delete it afterwards"""
+    xml_file = Path("data-processed.xml")
+    yield xml_file
+    xml_file.unlink(missing_ok=True)
+
+
+@pytest.mark.filterwarnings("ignore::UserWarning")
+def test_script_output(generated_xml_file: Path) -> None:
+    """Execute the import script and compare the generated XML with the expected XML"""
+    import_script.main()
+    with open("test/expected.xml", encoding="utf-8") as f:
+        xml_expected = _derandomize_xsd_id(f.read(), multiple_occurrences=True)
+    with open(generated_xml_file, encoding="utf-8") as f:
+        xml_returned = _derandomize_xsd_id(f.read(), multiple_occurrences=True)
+    assert _sort_xml_by_id(xml_expected) == _sort_xml_by_id(xml_returned)
```

Comment:
I'm not particularly a fan of this kind of testing against serializations. In this context it's ok, but I would generally rather try to test for contents than for serializations